python/ray/autoscaler/_private/cli_logger.py

```diff
@@ -301,6 +301,16 @@ def __init__(self):
         self._autodetected_cf_colormode = cf.colorful.colormode
         self.set_format()
 
+        self.callback_map = {}
+
+    def add_callback_handler(self, event, callback):
+        self.callback_map[event] = callback
+
+    def execute_callback(self, event, event_data):
+        if event in self.callback_map:
+            event_data["event_name"] = event
+            self.callback_map[event](event_data)
+
```

Comment:
Let's move this into `_private/event_system.py` and introduce a `EventSystem` class which ends up being defined as a global.

```python
# Don't introduce too many unnecessary imports
import typing
from enum import Enum

class Events(Enum):
    a = auto()
...

class _EventSystem:
    def __init__(self):
        self.callback_map = {}

    def add_callback_handler(self, event, callback):
        self.callback_map[event] = callback

    def execute_callback(self, event, event_data):
        if event in self.callback_map:
            event_data["event_name"] = event
            self.callback_map[event](event_data)

GlobalEventSystem = _EventSystem()
```
In every other class, just do:
```
from ray.autoscaler._private.event_system import GlobalEventSystem, Events
```

To get public Events enum do:
```python
# ray/autoscaler/sdk.py
from ray.autoscaler._private.event_system import Events
``` 
```
from ray.autoscaler.sdk import Events
```